go_utils/tests/photo_download_test.py

```diff
@@ -68,48 +70,168 @@
 ]
 
 
+@pytest.mark.photodownload
+@pytest.mark.downloadtest
+@pytest.mark.util
+def test_get_globe_photo_id():
+    bad_inputs = [None, "", "malformedURL"]
+    for input in bad_inputs:
+        null_output = get_globe_photo_id(input)
+        assert null_output is None, (
+            "None expected for input: " + input + ". Instead got: " + null_output
+        )
+
+
 @pytest.mark.photodownload
 @pytest.mark.util
 def test_bad_characters():
-    output = remove_bad_characters('<bad-/test|"\\filename:?>*')
-    assert output == "bad-testfilename"
+    assert remove_bad_characters(None) is None, "input: None; expected output: None"
+    assert remove_bad_characters("") == "", "input:''; expected output: ''"
+    assert remove_bad_characters('<bad-/test|"\\filename:?>*') == "bad-testfilename", (
+        "input:<bad-/test|\"\\filename:?>*; expected output: 'bad-testfilename'; Actual output"
+        + remove_bad_characters('<bad-/test|"\\filename:?>*')
+    )
+
+
+mhm_name_fields = [
+    "url_type",
+    "watersource",
+    "latitude",
+    "longitude",
+    "date_str",
+    "mhm_id",
+    "classification",
+]
+
+lc_name_fields = ["direction", "latitude", "longitude", "date_str", "lc_id"]
 
 
 @pytest.mark.photodownload
 @pytest.mark.parametrize(
-    "input_file, func, desired",
+    "input_file, func, excluded_fields, additional_info, desired",
     [
         (
             "go_utils/tests/sample_data/mhm_small.csv",
             get_mhm_download_targets,
-            desired_mhm_names,
+            [],
+            "",
```

Comment:
Why are you passing blank lists and strings?

Reply:
oo I wanted other tests to call this test function `test_all_field_naming`, so I passed blank lists (indicating no field should be excluded from the name) and "" (indicating no additional info was passed to the name). This allows me to test both extremes: including all fields, no fields, and some fields.

This test may slightly change if I restructure the API to ask users to specify what to Include, rather than exclude.